### DISCUSSION TITLE
extraModalFooterActions does not get current record value

### DISCUSSION THREAD

**Discussion by yashdesai87** [filamentphp/filament#11485] [Help]
### Package

Table builder

### Package Version

3.0

### How can we help you?

I'm trying to create additional modal actions inside a modal. However, I'm not able to get the ID of the record that is triggering the modal. It retains the value of the first button clicked and reuses it even if another record is clicked.

     public static function table(Table $table): Table
    {
    	return $table->actions([
    		Action::make('action')
    			->label('Take Action')
    			->button()
    			->extraModalFooterActions(function(Action $action){
    				$ticket = $action->getRecord();
    				dump($ticket->id);
    				return [];
    			})
    	);
    }

The ticket ID of the first record clicked is being shown after every other trigger button is clicked. The ID does not change dynamically depending on the button clicked.

**Comment by clinvest** [comment]
Is this working?

```
public static function table(Table $table): Table
{
	return $table->actions([
		Action::make('action')
			->label('Take Action')
			->button()
			->extraModalFooterActions(function(Action $action, $record){
				$ticket = $record;
				dump($ticket->id);
				return [];
			})
	);
}
```

**Comment by yashdesai87** [comment]
> Is this working?
> 
> ```
> public static function table(Table $table): Table
> {
> 	return $table->actions([
> 		Action::make('action')
> 			->label('Take Action')
> 			->button()
> 			->extraModalFooterActions(function(Action $action, $record){
> 				$ticket = $record;
> 				dump($ticket->id);
> 				return [];
> 			})
> 	);
> }
> ```

No, it doesn't. Still retains the ID of the last record I clicked.

**Reply by leandrocfe** [reply]
You should inject `$record`. Please share the whole code that you are trying to do
